application/org.openjdk.jmc.flightrecorder.controlpanel.ui/src/main/java/org/openjdk/jmc/flightrecorder/controlpanel/ui/ControlPanel.java

```diff
@@ -123,9 +123,8 @@ public static MCFile getDefaultRecordingFile(String suffixPart) {
 				? suffixPart.substring(0, MAX_FILENAME_SUFFIX_LENGTH) : suffixPart;
 
 		String lastPathStr = getDefault().getPreferenceStore().getString(LAST_PATH);
-		IPath lastFolder = lastPathStr.isEmpty() ?
-				IDESupportToolkit.getDefaultRecordingFolder() :
-				Path.fromOSString(lastPathStr).removeLastSegments(1);
+		IPath lastFolder = lastPathStr.isEmpty() ? IDESupportToolkit.getDefaultRecordingFolder()
+				: Path.fromOSString(lastPathStr).removeLastSegments(1);
```

Comment:
I actually kind of like this one. Can spotless be configured to allow separating ternary operations like the old style? If they can't be on one line or three then maybe they should be expanded to if/else statements? Having it on just two lines makes it a little bit harder to see what's going on.

Reply:
There are a few cases where the formatting gets a bit worse, agreed. Not too many, and it usually has to do with line breaks. I think the two choices for these are to either make the formatter weaker (i.e. not touch explicit line breaks the user has entered), or increase the maximum line length.

Reply:
I think I can live with this one. Let me know if you disagree!

Reply:
I can live with that one, but the matrices are a bit harder to accept.

Reply:
They should now be fixed.